stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1164,12 +1164,64 @@ func _TFCGetGlobalEagerContext() -> CTFEContext {
 // Some of the functions are marked with @silgen_name instead of @_cdecl,
 // because their input/output data types are not C-compatible
 // (e.g. AnyTensorHandle).
-@inlinable
-@_silgen_name("_swift_tfc_GetCTensorHandleFromSwift")
-public func _TFCGetCTensorHandleFromSwift(
-  _ handle: _AnyTensorHandle
-) -> CTensorHandle {
-  return handle.cTensorHandle
+
+/// Adds `handle` as an input to `op`.
+@usableFromInline
+@_silgen_name("_swift_tfc_OpAddInputFromTensorHandle")
+func _TFCOpAddInputFromTensorHandle(_ op: CTFEOp,
+                                    _ handle: _AnyTensorHandle,
+                                    _ status: CTFStatus) {
+  TFE_OpAddInput(op, handle._cTensorHandle, status)
+}
+
+/// Adds `t` as an input or inputs to `op`. Returns the number of inputs added.
+@usableFromInline
+@_silgen_name("_swift_tfc_OpAddInputFromTensorGroup")
+func _TFCOpAddInputFromTensorGroup<T : TensorGroup>(
+    _ op: CTFEOp, _ t: T, _ status: CTFStatus
+) -> Int32 {
+  let count = Int(T._tensorHandleCount)
+  let buffer =
+      UnsafeMutableBufferPointer<CTensorHandle>.allocate(capacity: count)
+  defer { buffer.deallocate() }
+  t._unpackTensorHandles(into: buffer.baseAddress!)
```

Comment:
Please add a test for an empty struct, in which case `baseAddress` may be `nil`.

Reply:
good catch. I added a test and it seems to pass, but I'll change `_unpackTensorHandles` to take an `UnsafeMutablePointer?` anyways, so that we only do force-unwrapping when we're sure that the struct is nonempty.